src/main/java/com/geronimo/model/AuditedEntity.java

```diff
@@ -0,0 +1,25 @@
+package com.geronimo.model;
+
+import lombok.Setter;
+import lombok.ToString;
+import org.springframework.data.annotation.CreatedDate;
+import org.springframework.data.annotation.LastModifiedDate;
+import org.springframework.data.jpa.domain.support.AuditingEntityListener;
+
+import javax.persistence.EntityListeners;
+import javax.persistence.MappedSuperclass;
+import java.util.Date;
+
+@Setter
+@ToString
+@MappedSuperclass
+@EntityListeners(AuditingEntityListener.class)
+abstract class AuditedEntity extends AbstractEntity {
```

Comment:
By the way, what about getters? Setter & Getter annotations maybe? 

Also entities need to be comparable by content so we need equals & hashCode, right?
There's an annotation of lombok for that, you'll need to annotate super classes as well, and the classes that extend them should also have this annotation with (callSuper = true) to take super-class methods into consideration.

Reply:
Data annotation does everything as far as I know: setters, getters, equals&hashcode, toString.